x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/dimension_container.tsx

```diff
@@ -32,8 +32,11 @@ export function DimensionContainer({
   const [focusTrapIsEnabled, setFocusTrapIsEnabled] = useState(false);
 
   const closeFlyout = () => {
-    handleClose();
-    setFocusTrapIsEnabled(false);
+    // some internal popovers are pretty slow to unmount, so give it some time
+    setTimeout(() => {
+      handleClose();
+      setFocusTrapIsEnabled(false);
+    }, 150); // <= 150 has been chosen from empirical testing
```

Comment:
I'm really concerned about this approach. As you said it will add a delay as well as not even work in all cases depending on the timing of things.

I think I wasn't really clear in my original review, I think it's fine to use the updater function `setState` approach to make sure the flyout stays closed if the additional setStates triggered on closing the popover  are really necessary, but I was under the impression the state update happening in this scenario doesn't even make sense and we can just avoid it all together.

So, in summary
* Let's please not introduce constant waiting times in the UI if avoidable
* See if it's possible to not update the state on unmount like this
* If we have to, use the updater function to make sure the flyout stays closed, but limit the cases where this is happening

Reply:
I'll move the issue upstream and have EUI fix it at component level.

Reply:
Created issue https://github.com/elastic/eui/issues/4265 on EUI repo and will remove the code here.